handout/architect/architect-guide-source-code-to-working-software.md

```diff
@@ -0,0 +1,128 @@
+also note that Java does not solve the _distribution_ aspect of the problem
+- Note that the web existed and had some basic capabilities and that it
+_evolved_ into an application stack that was not just portable in the 
+traditional sense but also inherently distributable, and fortunately ubiquitous.
+
+This would ideally be about three minutes since it could be distilled in a
+handful of sentences.
+
+
+### The Problem Angular and TypeScript Solve
+
+- The browser and specifically the DOM were designed to render _documents_ not
+programs but documents
+- Working _efficiently_ with the DOM to replicate "native" software look/feel
+has challenges
+- Efficiently organizing source code in a language that was created in ten days
+is non-trivial, consequently harder to maintain, teach, etc
+- Efficiently organizing the aforementioned source code along with the two DOM
+languages - HTML and CSS - is also challenging
+- Maintaining the tools necessary to orchestrate all of these things is yet
+another task unto itself
+
+This would also ideally be about five minutes
+
+
+## Act II - Making Things Happen
+
+### From Code To Runnable Code
+
+Starting sample code: _Working angular environment with zero moving parts a
+single component "Hello World" will do_
+
+- Quickly walk through the sample code to highlight all of the TypeScript code
+and the references to Angular library code
+- Run TypeScript to produce a regular transpilation (individual files, not a 
+bundle)
+- Compare the input source to the ES5 code that will actually be executed and
+note the decorators are just functions but don't get into details
+
+This would ideally be a five minute procedure
+
+
+### From Runnable Code To The User
+
+- Describe at a very specific but high level the physical networking and wires, 
+TCP, HTTP, DNS
+- Enter the web server
+- Run the code with live server and marvel at "Hello World"
+- Build/Run/Execute a specially crafted Hello World that depends on an Angular
+lib marked with `debugger` statements (to facilitate fast iteration)
+- During this debugger task stop at the specific phases of what is going on, 
+do not get bogged down into the dark corners
+
+This would take about ten minutes
+
+### From The User To Runnable Code
+
+- Add an input field and an output area to the code, build, and run it using the
+`debugger` backed angular source
+- Step through typing in a single character specifically noting angular's 
+phases
+- Note the event loop
+
+This could be quick but it's hard to imagine squeezing it into five minutes
+
+
+## Act III - That's Not All Folks
+
+### "Production" Code
+
+- AoT and how it changes the source
+- Step through a pre-baked copy of Hello World with AoT'd source
+- Look at the size of bundled code and note the delivery challenges
+- Very abstract description of code splitting with `NgModule`
+- Abstract description of SSR, perceived load times, and SEO 
```

Comment:
I think this can be a reference to another chapter - I'm hoping to talk someone into doing something on SSR.